### PR TITLE
Incomplete Form Listing and Opening

### PR DESCRIPTION
**Pull request by wpride**
Adds the front end for listing and selecting incomplete form sessions. Screen shots attached. Obvious UI TODOs:
- The list just shows the Session ID - would need to start storing the form title with the sessions
- The font awesome icon won't scale to a reasonable size

Cross request: https://github.com/dimagi/formplayer/pull/50
Contains: https://github.com/dimagi/commcare-hq/pull/12186

screen shot 2016-06-28 at 6 43 58 pm
screen shot 2016-06-28 at 6 43 50 pm

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/entities/sessions.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                    options.data = JSON.stringify({
+                        "sessionId": sessionId,
+                    });
+
+                    options.url = formplayerUrl + '/incomplete-form';
+                    options.type = 'POST';
+                    options.dataType = "json";
+                    options.crossDomain = {crossDomain: true};
+                    options.xhrFields = {withCredentials: true};
+                    options.contentType = "application/json";
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/entities/sessions.js]: could probably abstract this into a utility function `getCrossDomainAjaxOptions`

**Comment by benrudolph** [comment]
👍 📝

**Comment by wpride** [comment]
@benrudolph updated UI below - just increased the font size and added some shadowing. Please let me know if there's better best practices (or just general comments) @biyeun 

screen shot 2016-07-01 at 12 22 03 pm

File: corehq/apps/cloudcare/templates/formplayer/grid_view.html [hunk 2]
```diff
@@ -6,12 +6,20 @@
...
+        </div>
 
     </div>
 </script>
 
 <script id="row-template" type="text/template">
-        <h3><%- name %></h3>
-        <image class="app-icon" src="{% static 'cloudcare/images/cc_flower.png' %}"/>
+    <h3><%- name %></h3>
+    <image class="app-icon" src="{% static 'cloudcare/images/cc_flower.png' %}"/>
```
Comment (biyeun) [inline corehq/apps/cloudcare/templates/formplayer/grid_view.html]: `img`

Comment (benrudolph) [inline corehq/apps/cloudcare/templates/formplayer/grid_view.html]: haha did this work for you @wpride? maybe chrome automatically converts it

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/util/util.js [hunk 3]
```diff
@@ -33,4 +33,12 @@ Util.getSteps = function (queryString) {
     }
     paramMap.steps = steps;
     return paramMap;
+};
+
+Util.setCrossDomainAjaxOptions = function(options) {
```
Comment (benrudolph) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/util/util.js]: 👍

File: corehq/apps/cloudcare/templates/formplayer/session_list.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    <p><h2>Sessions</h2></p>
+    <table class="table table-hover table-striped table-bordered">
+        <tbody>
+        </tbody>
+    </table>
+</script>
+
+<script type="text/template" id="session-view-item-template">
+    <td class="col-sm-2">
+       <div>ID: <%- sessionId %> </div>
```
Comment (benrudolph) [inline corehq/apps/cloudcare/templates/formplayer/session_list.html]: does it make sense for this to be user facing? can we show a name instead?

Comment (benrudolph) [inline corehq/apps/cloudcare/templates/formplayer/session_list.html]: nvm just read your comment on it

**Pull request merged**